spyder/workers/tests/test_update.py

```diff
@@ -6,56 +6,133 @@
 
 import pytest
 
-from spyder.config.utils import is_anaconda
 from spyder.workers.updates import WorkerUpdates
 
 
-def test_update(qtbot):
-    """Test we offer updates for lower versions."""
+@pytest.mark.parametrize("is_anaconda", [True, False])
+@pytest.mark.parametrize("is_pypi", [True, False])
+@pytest.mark.parametrize(
+    "spyder_conda_channel", [
+        ("pkgs/main", "https://repo.anaconda.com/pkgs/main"),
+        ("conda-forge", "https://conda.anaconda.org/conda-forge")
+    ]
+)
+def test_update(qtbot, mocker, is_anaconda, is_pypi, spyder_conda_channel):
+    """Test we offer updates for Anaconda and PyPI."""
+    mocker.patch(
+        "spyder.workers.updates.is_anaconda",
+        return_value=is_anaconda
+    )
+
+    if is_anaconda:
+        if is_pypi:
+            channel = ("pypi", "https://conda.anaconda.org/pypi")
+        else:
+            channel = spyder_conda_channel
+
+        mocker.patch(
+            "spyder.workers.updates.get_spyder_conda_channel",
+            return_value=channel
+        )
+
+    worker = WorkerUpdates(None, False, version="1.0.0")
+    worker.start()
+    assert worker.update_available
+    assert len(worker.releases) == 1
+
+
+@pytest.mark.parametrize("is_anaconda", [True, False])
+@pytest.mark.parametrize("is_pypi", [True, False])
+@pytest.mark.parametrize(
+    "spyder_conda_channel", [
+        ("pkgs/main", "https://repo.anaconda.com/pkgs/main"),
+        ("conda-forge", "https://conda.anaconda.org/conda-forge")
+    ]
+)
+def test_no_update(qtbot, mocker, is_anaconda, is_pypi, spyder_conda_channel):
```

Comment:
Should we make `version` a pytest parameter in `test_update` so that we don't have to duplicate code here? `test_update` and `test_no_update` have exactly the same code except for the version string and the assertion of `worker.update_available`.

Reply:
Very good point, thanks @mrclary! I amended the commit where I expanded this test to include your suggestion.